packages/grafana-ui/src/components/Graph/Graph.tsx

```diff
@@ -179,6 +202,70 @@ export class Graph extends PureComponent<GraphProps, GraphState> {
     });
   };
 
+  renderContext = () => {
+    const { series } = this.props;
+    const { contextPos, contextItem, isContextVisible } = this.state;
+
+    if (!isContextVisible || !contextPos || !contextItem || series.length === 0) {
+      return null;
+    }
+
+    // Indicates column(field) index in y-axis dimension
+    const seriesIndex = contextItem ? contextItem.series.seriesIndex : 0;
+    // Indicates row index in context field values
+    const rowIndex = contextItem ? contextItem.dataIndex : undefined;
+
+    const contextDimensions: ContextDimensions<GraphDimensions> = {
+      // Described x-axis context item
+      xAxis: [seriesIndex, rowIndex],
+      // Describes y-axis context item
+      yAxis: contextItem ? [contextItem.series.seriesIndex, contextItem.dataIndex] : null,
+    };
+
+    const dimensions: GraphDimensions = {
+      // time/value dimension columns are index-aligned - see getGraphSeriesModel
+      xAxis: createDimension(
+        'xAxis',
+        series.map(s => s.timeField)
+      ),
+      yAxis: createDimension(
+        'yAxis',
+        series.map(s => s.valueField)
+      ),
+    };
```

Comment:
Reused from renderTooltip(). We are creating and passing dimensions for GraphContextMenu component as well, so we can use dimensions to calculate the value, that is going to be rendered and that is going to be consistent with Tooltip. I wasn't sure, if better approach would be to calculate the value here, and then just pass it to `GraphContextMenu`, but I've decided for this approach, as the same approach is used for Tooltip.

Reply:
Having the same pattern as they are similar things makes sense here.

Reply:
Yeah, good choice. At some point we need to unify this tho, so that code isn't duplicated